build.proj

```diff
@@ -2,6 +2,16 @@
 <Project ToolsVersion="14.0" DefaultTargets="BuildTheWholeCli" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <Import Project="build/Microsoft.DotNet.Cli.Monikers.props" />
+  <ImportGroup Condition=" '$(GeneratingPropsFile)' == 'true' ">
+    <Import Condition=" Exists('$(GitCommitInfoProps)') "
+            Project="$(GitCommitInfoProps)" />
+    <Import Condition=" Exists('$(HostInfoProps)') "
+            Project="$(HostInfoProps)" />
+  </ImportGroup>
+  <ImportGroup Condition=" '$(GeneratingPropsFile)' != 'true' ">
+    <Import Project="$(GitCommitInfoProps)" />
+    <Import Project="$(HostInfoProps)" />
+  </ImportGroup>
 
```

Comment:
Actually addressing a later comment you have would allow me to move this to dir.props